server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -783,9 +784,16 @@ protected Node(
                 threadPool
             );
 
-            final SearchRequestStats searchRequestStats = new SearchRequestStats();
+            final SearchRequestStats searchRequestStats = new SearchRequestStats(clusterService);
             final SearchRequestSlowLog searchRequestSlowLog = new SearchRequestSlowLog(clusterService);
 
+            // register all standard SearchRequestOperationsListeners to the SearchRequestListenerManager
+            final SearchRequestListenerManager searchRequestListenerManager = new SearchRequestListenerManager(clusterService);
+            searchRequestListenerManager.addListeners(
+                searchRequestStats,
+                searchRequestSlowLog
+            );
```

Comment:
Since you create the listeners before creating the listener manager, do you even need an `addListeners` (or `removeListeners`) method?

Once you have a plugin, you just need to make sure that the listeners provided by plugins are created before you create the manager. Then you don't need to add/remove listeners at runtime.

Reply:
Good point. refactored the code to remove `addListeners` and `removeListeners`. All listeners would be added to the manager in the constructor.